pkg/controller/registry/resolver/source_csvs.go

```diff
@@ -85,6 +93,20 @@ func (s *csvSource) Snapshot(ctx context.Context) (*cache.Snapshot, error) {
 			continue
 		}
 
+		if cachedSubscription, ok := csvSubscriptions[csv]; !ok || cachedSubscription == nil {
+			// we might be in an incoherent state, so let's check with live clients to make sure
+			realSubscriptions, err := s.listSubscriptions(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to list subscriptions: %w", err)
+			}
+			for _, realSubscription := range realSubscriptions.Items {
+				if realSubscription.Status.InstalledCSV == csv.Name {
+					// oops, live cluster state is coherent
```

Comment:
```suggestion
					// oops, live cluster state is incoherent
```

Reply:
That's not correct - if we're here, the live cluster state is coherent - i.e. we found that a subscription really does exist with the CSV in status.

Reply:
That feels weird that the comment says if we got here the cluster is coherent, but the error message immediately following it says the lister caches are incoherent.  
Isn't the mismatch our basis for determining that the cluster state is incoherent?  (i.e. would the live listers ever be wrong, as suggested by the error below?)

Reply:
The cluster - and the live `List()` calls - cannot, by definition, be incoherent. Our *caches* are incoherent, which is what the error message says.